### PR TITLE
Migrations+Triggers

### PR DESCRIPTION
**Pull request by AlexandruBaris**

### CODE REVIEW

File: src/main/resources/db/migration/V20191004032400__create_table_locations.sql [hunk 1]
```diff
@@ -5,7 +5,8 @@ CREATE TABLE locations
     postal_code    VARCHAR2(12),
     city           VARCHAR2(30),
     state_province VARCHAR2(25),
-    country_id     CHAR(2) NOT NULL
+    country_id     CHAR(2) NOT NULL,
+    department_amount NUMBER(3)
```
Comment (codreanvs) [inline src/main/resources/db/migration/V20191004032400__create_table_locations.sql]: Please create a separated migration to add this new column.

Comment (AlexandruBaris) [reply src/main/resources/db/migration/V20191004032400__create_table_locations.sql]: created

File: homework-selects/sql_questions.sql [hunk 2]
```diff
@@ -3,56 +3,419 @@
 SELECT first_name, last_name, salary, job_title
 FROM employees
          LEFT JOIN jobs USING (job_id);
+
 -- 2. the first and last name, department, city, and state province for each employee.
+SELECT A.first_name, A.last_name, B.department_name, C.city, C.state_province
+FROM employees A
+         JOIN DEPARTMENTS B
+            ON A.DEPARTMENT_ID = B.DEPARTMENT_ID
```
Comment (vermurachi) [inline homework-selects/sql_questions.sql]: You could use LEFT JOIN ... USING like in the first example to make the commands shorter

Comment (vermurachi) [inline homework-selects/sql_questions.sql]: You could use LEFT JOIN ... USING like in the first example to make the commands shorter

File: homework-selects/sql_questions.sql [hunk 3]
```diff
@@ -3,56 +3,419 @@
...
 --10. the department name and number of employees in each of the department.
+SELECT DEPARTMENT_NAME, COUNT(*) AS "Number of employees"
+FROM DEPARTMENTS
+         JOIN EMPLOYEES ON EMPLOYEES.DEPARTMENT_ID=Departments.DEPARTMENT_ID
+GROUP BY EMPLOYEES.DEPARTMENT_ID, DEPARTMENT_NAME;
+
 --11. the name of the department, average salary and number of employees working in that department who got commission.
+SELECT A.DEPARTMENT_NAME, AVG(B.SALARY),COUNT(COMMISSION_PCT)
+FROM EMPLOYEES B
+         JOIN DEPARTMENTS A ON B.DEPARTMENT_ID = A.DEPARTMENT_ID
```
Comment (vermurachi) [inline homework-selects/sql_questions.sql]: where do you check the employees who got commission?

Comment (vermurachi) [inline homework-selects/sql_questions.sql]: where do you check the employees who got commission?

File: homework-selects/sql_questions.sql [hunk 4]
```diff
@@ -3,56 +3,419 @@
...
 --26. the employee number, name( first name and last name ), and salary for all employees who earn more than the average salary and who work in a department with any employee with a J in their name.
+SELECT EMPLOYEE_ID, FIRST_NAME, LAST_NAME, SALARY
+FROM EMPLOYEES
+WHERE SALARY > (
+    SELECT AVG(SALARY)
+    FROM EMPLOYEES
+) AND DEPARTMENT_ID IN (
+    SELECT DEPARTMENT_ID
+    FROM EMPLOYEES
+    WHERE FIRST_NAME LIKE '%J'
```
Comment (vermurachi) [inline homework-selects/sql_questions.sql]: This will select only the names that ends with 'J'

Comment (vermurachi) [inline homework-selects/sql_questions.sql]: This will select only the names that ends with 'J'

File: homework-selects/sql_questions.sql [hunk 5]
```diff
@@ -3,56 +3,419 @@
...
+    SELECT SALARY
+    FROM EMPLOYEES
+    WHERE DEPARTMENT_ID =40
+);
+
 --44. the first and last name, salary, and department ID for those employees who earn more than the minimum salary of a department which ID is 40.
+SELECT FIRST_NAME, LAST_NAME, SALARY, DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE SALARY > ANY(
+    SELECT SALARY
```
Comment (vermurachi) [inline homework-selects/sql_questions.sql]: Where do you check what is the minimum salary?

Comment (vermurachi) [inline homework-selects/sql_questions.sql]: Where do you check what is the minimum salary?

**Comment by vermurachi** [comment]
Try to keep SQL commands in UPPERCASE and table_names and column_names in lower for readability

File: homework-selects/sql_questions.sql (hunk 4 above)
Comment (AlexandruBaris) [reply homework-selects/sql_questions.sql]: changed %J%

File: homework-selects/sql_questions.sql (hunk 5 above)
Comment (AlexandruBaris) [reply homework-selects/sql_questions.sql]: MIN ( SALARY)

**Review by vermurachi — APPROVED** [review]

**Pull request merged**